py/server/deephaven/_wrapper.py

```diff
@@ -162,15 +163,35 @@ def pythonify(j_obj: Any) -> Optional[Any]:
     return wrap_j_object(j_obj)
 
 
+def _wrap_with_subclass(j_obj: jpy.JType, cls: type) -> Optional[JObjectWrapper]:
+    """ Returns a wrapper instance for the specified Java object if the specified class is a subclass of the wrapper
+    class. Otherwise, returns None. """
+    subclasses = cls.__subclasses__()
+    for subclass in subclasses:
+        try:
+            if (wrapper := _wrap_with_subclass(j_obj, subclass)) is not None:
+                return wrapper
+            return subclass(j_obj)
+        except:
+            continue
+    return None
+
 def wrap_j_object(j_obj: jpy.JType) -> Union[JObjectWrapper, jpy.JType]:
```

Comment:
Here is some concrete code that illustrates what a mess this can quickly become:

```python
class A:
    def __init__(self):
        self.a = 1

class B(A):
    def __init__(self):
        self.b = 1

class C:
    def __init__(self):
        self.c = 1

class D(A,C):
    def __init__(self):
        self.d = 1

class E(B,A):
    def __init__(self):
        self.e = 1

class F(B):
    def __init__(self):
        self.f = 1

print(A.__subclasses__())
print(B.__subclasses__())
print(C.__subclasses__())
print(D.__subclasses__())
print(E.__subclasses__())
print(F.__subclasses__())
```
```
[<class '__main__.B'>, <class '__main__.D'>, <class '__main__.E'>]
[<class '__main__.E'>, <class '__main__.F'>]
[<class '__main__.D'>]
[]
[]
[]
```

Reply:
The example makes sense if we are dealing with generic, arbitrary class hierarchy. Yet in our specific case, as long as we make sure (and we must) that all the subclasses are distinct from each other and their uniqueness is checked at instance initialization. Then we are guaranteed to the wrapper class that best matches what the Java object is.

Reply:
I think what is here will work for now.  There is a burden on future reviewers to make certain that there is uniqueness in the constructors.